website/docs/more/deployment.mdx

```diff
@@ -0,0 +1,43 @@
+
+:::important A note about `trunk serve --release`
+Do **not** use `trunk serve --release`  to serve your application in production. It should only be used for testing the release build during development
+:::
+
+## Server configuration
+
+### Serving `index.html` as fallback
+
+If the application uses the [Yew router](../../router), you must configure the server to return the `index.html` when asked for a file that it does not have.
+
+An application with Yew router is built as a [Single Page Application (SPA)](https://developer.mozilla.org/en-US/docs/Glossary/SPA). When the user navigates to a URL from within a running client, the router interprets the URL and routes to that page. 
+
+But on a fresh load, such as when navigating to the page by entering it in the address bar or refreshing the page, all of these actions are handled by the browser itself, outside the running application. The browser makes a direct request to the server for that URL, bypassing the router. The server sees that no route is matched and returns 404.
+
+A server normally returns `index.html` when it receives a request for `/`, but it rejects `/show/42` and returns a 404 - Not Found error unless it is configured to return `index.html` instead. By returning `index.html`, we load the app as it would normally load if request was for `/`, the router notices that the the route is `/show/42` and displays the appropriate contents.
+
+### Configuring correct MIME-type for Web Assembly asset.
+
+The WASM files must be served with the [Content-Type header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type) set to `application/wasm` MIME-type.
+
+Most servers and hosting services already do this by default. If yours doesn't, consult it's documentation. Incorrect MIME-type will result in a following error:
+```
+`WebAssembly.instantiateStreaming` failed because your server does not serve wasm with `application/wasm` MIME type. Falling back to `WebAssembly.instantiate` which is slower. Original error:
+ TypeError: WebAssembly: Response has unsupported MIME type 'text/plain' expected 'application/wasm'
+```
+
+## Building for Relative Paths
+
+By default, trunk will assume that your site is being served at `/` and build the site accordingly. This behavior can be overridden by adding `<base data-trunk-public-url />` to the `index.html` file. Trunk rewrites this tag to contain the value passed to `--public-url`.. Yew router automatically detects the presence of `<base />` and handles it appropriately.
+
+## Customizing behavior using environment variables
+
+It's common to customize the build environment by using environment variables. Since the app is run in browser, we can't read the environment variables at compile time.
```

Comment:
```suggestion
It's common to customize the build environment by using environment variables. Since the app is run in browser, we can't read the environment variables at runtime.
```